packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-table.ts

```diff
@@ -0,0 +1,311 @@
+   *
+   * @default Duration.hours(1)
+   */
+  readonly retryTimeout?: cdk.Duration;
+
+  /**
+   * The intermediate bucket where Firehose will stage your data before COPYing it to the Redshift cluster.
+   *
+   * @default - a bucket will be created for you.
+   */
+  readonly intermediateBucket?: s3.IBucket;
+
+  /**
+   * The role that is attached to the Redshift cluster and will have permissions to access the intermediate bucket.
+   *
+   * If a role is provided, it must be already attached to the cluster, to avoid the 10 role per cluster limit.
+   *
+   * @default - a role will be created for you.
+   */
+  readonly bucketAccessRole?: iam.IRole;
+}
+
+/**
+ * Redshift table delivery stream destination.
+ */
+export class RedshiftTable implements firehose.IDestination {
+  /**
+   * The Redshift database that contains the table were data will be delivered.
+   */
+  public readonly database: string;
+
+  /**
+   * The name of the Redshift table were data will be delivered.
+   */
+  get tableName(): string {
+    if (!this._tableName) {
+      throw new Error('Bind must be called before the name of the generated table is available');
+    }
+    return this._tableName;
+  }
+
+  private _tableName?: string;
+  private readonly adminUser?: secretsmanager.ISecret;
+
+  constructor(private readonly cluster: redshift.ICluster, private readonly props: RedshiftDestinationProps) {
+    if (!cluster.publiclyAccessible) {
+      throw new Error('Redshift cluster used as delivery stream destination is not publicly accessible');
+    }
+    if (!cluster.subnetGroup?.selectedSubnets?.hasPublic) {
+      throw new Error('Redshift cluster used as delivery stream destination is not located in a public subnet');
+    }
+    if (props.retryTimeout && props.retryTimeout.toSeconds() > cdk.Duration.hours(2).toSeconds()) {
+      throw new Error(`Retry timeout must be less than 7,200 seconds (2 hours), given ${props.retryTimeout.toSeconds()} seconds`);
+    }
+    if (props.compression === Compression.SNAPPY || props.compression === Compression.HADOOP_SNAPPY || props.compression === Compression.ZIP) {
+      throw new Error(`Redshift delivery stream destination does not support ${props.compression.value} compression`);
+    }
+    if (props.s3Backup?.mode === BackupMode.FAILED) {
+      throw new Error('Redshift delivery stream destination does not support ${props.s3Backup?.mode} backup mode');
+    }
+
+    const generateTable = !props.tableName;
+    const generateUser = !props.user.username || !props.user.password;
+    const adminUser = props.adminUser ?? (cluster instanceof redshift.Cluster ? cluster.secret : undefined);
+    if (!adminUser && (generateTable || generateUser)) {
+      throw new Error(
+        'Administrative access to the cluster is required but an admin user secret was not provided and either the Redshift cluster did not generate admin user credentials (they were provided explicitly) or the cluster was imported',
+      );
```

Comment:
This is a doozy of an error message. But I think you get the point across. Maybe consider splitting it up? This suggestion is not completely correct, but hopefully you get the idea. 

```suggestion
if (!(cluster instanceof redshift.Cluster)) {
      throw new Error(
        'Administrative access to the cluster is required but an admin user secret was not provided and the cluster was imported',
      );
}
if (!cluster.secret) {
      throw new Error(
        'Administrative access to the cluster is required but an admin user secret was not provided and the Redshift cluster did not generate admin user credentials (they were provided explicitly)',
      );
}  
```